### DISCUSSION TITLE
Using Audio Gridder on one system

### DISCUSSION THREAD

**Discussion by BubsonGH** [apohl79/audiogridder#654] [General]
Hi,

Just wanted to share a test I did today just in case this has not been attempted yet.

I am running a 5950x CPU, as powerful as it is in the realm of running many tracks I can still run into issues with single core performance, primarily on my master bus when stacking up some acustica audio plugins! 

My personal preferred work flow is to do everything in one project and avoid bouncing and freezing tracks.. I prefer to have everything there ready to adjust as needed. So.. as you can imagine my CPU usage by the time I get to mastering can be bouncing around 40 - 60% + (not to mention that I like to get some top down mixing going too)

Taking into account that once I start populating my master channel with plugins from acustica.. my project can start choking pretty hard.

So...

today I decided to grab audiogridder and see if I could run it all on one system, hoping to outsource specifically my master channel only.

I installed the components on one pc system and replaced my master chain with separate audio gridder wrappers for each plugin.

It works! 

As far as I can tell.. due to sending each plugin out on its own AG instance, the master chain is now being spread across different cores of my system and leaving my studio one project with much more breathing room!

My master bus if all plugins were turned on before could completely choke even an empty project, but now there is zero usage in the daw, outside the daw via Audio Gridder server is only using about 15% of my CPU.

Couple things I would like to see for my personal setup...

I have of course turned off the video streaming since I am on one system, and I am using 1 instance of AG per plugin.
With that in mind I would like to have the option maybe as a tick box to have AG open the single plugin running inside it automatically when I double clicked the AG instance in my master chain, just to save time having to double click AG instance then click the plugin to open it in the wrapper.

Secondly I am guessing the plugin automatically opens top left of the screen every time for the sake of video capturing it, would be a nice touch to have the option to just have the plugin remember where you last left it on screen and reopen it in the save place. This way in my DAW template I could just setup as I want it.

Maybe all of this could come under a local mode section that you can activate?

One other thing is that I found trying to run all the master bus plugins in one instance of AG seemed to work but if I tried to save that as a preset in Studio Ones system, trying to then load it seemed to crash Audio gridder every time, so running separate instances and then saving as an effects chain preset seems to work better.

Just some suggestions.. Great work so far! some of my projects can breath again!

Dan

**Comment by CaBuffalo** [comment]
Hi, BubsonGH,

Thanks for the great feedback. We are glad to see AG works for your project
nicely.

To answer your questions:

1, I recommend you to set up a macro or shortcut to open / close the plugin
in S1. It's a better way to speed up your workflow.
2, There are other uses mentioned as well. We will consider it in the
future.
3, I have set up a preset with 3 plugins in one AG instance. It works well.
Not with more than 3 so far.

On Tue, Jul 20, 2021 at 10:38 AM BubsonGH ***@***.***> wrote:

> Hi,
>
> Just wanted to share a test I did today just in case this has not been
> attempted yet.
>
> I am running a 5950x CPU, as powerful as it is in the realm of running
> many tracks I can still run into issues with single core performance,
> primarily on my master bus when stacking up some acustica audio plugins!
>
> My personal preferred work flow is to do everything in one project and
> avoid bouncing and freezing tracks.. I prefer to have everything there
> ready to adjust as needed. So.. as you can imagine my CPU usage by the time
> I get to mastering can be bouncing around 40 - 60% + (not to mention that I
> like to get some top down mixing going too)
>
> Taking into account that once I start populating my master channel with
> plugins from acustica.. my project can start choking pretty hard.
>
> So...
>
> today I decided to grab audiogridder and see if I could run it all on one
> system, hoping to outsource specifically my master channel only.
>
> I installed the components on one pc system and replaced my master chain
> with separate audio gridder wrappers for each plugin.
>
> It works!
>
> As far as I can tell.. due to sending each plugin out on its own AG
> instance, the master chain is now being spread across different cores of my
> system and leaving my studio one project with much more breathing room!
>
> My master bus if all plugins were turned on before could completely choke
> even an empty project, but now there is zero usage in the daw, outside the
> daw via Audio Gridder server is only using about 15% of my CPU.
>
> Couple things I would like to see for my personal setup...
>
> I have of course turned off the video streaming since I am on one system,
> and I am using 1 instance of AG per plugin.
> With that in mind I would like to have the option maybe as a tick box to
> have AG open the single plugin running inside it automatically when I
> double clicked the AG instance in my master chain, just to save time having
> to double click AG instance then click the plugin to open it in the wrapper.
>
> Secondly I am guessing the plugin automatically opens top left of the
> screen every time for the sake of video capturing it, would be a nice touch
> to have the option to just have the plugin remember where you last left it
> on screen and reopen it in the save place. This way in my DAW template I
> could just setup as I want it.
>
> Maybe all of this could come under a local mode section that you can
> activate?
>
> One other thing is that I found trying to run all the master bus plugins
> in one instance of AG seemed to work but if I tried to save that as a
> preset in Studio Ones system, trying to then load it seemed to crash Audio
> gridder every time, so running separate instances and then saving as an
> effects chain preset seems to work better.
>
> Just some suggestions.. Great work so far! some of my projects can breath
> again!
>
> Dan
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/654>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6BJ6A62JOYIY4WOJ5DTYWYCDANCNFSM5AWIEUYA>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Reply by BubsonGH** [reply]
Sound great, I look forward to seeing how the tool is built out, Ill continue testing under my setup and feedback any issues I might run into.

Many thanks, Audio Gridder is a life saver.

**Reply by BubsonGH** [reply]
Quick question, I have the plugin Cassette by WavesFactory. I can see it was scanned and found by the server under available plugins but it is not showing in the plugin search, Any ideas why?

Thanks

**Reply by BubsonGH** [reply]
Never mind I just tried disabling server filter and it showed up, Thanks

**Reply by BubsonGH** [reply]
By the way, I notice audio gridder supports the ability to look up the plugins presets. 

I dont know if it is possible to do, but if you could make audio gridder see the presets that you have created via studio one that would be great. I dont save presets via plugins specifically anymore since it is so convenient via studio one to just save them at the DAW level so you never have to back them up etc.

Thanks

**Comment by apohl79** [comment]
Sounds like you are using 1.1 stable? You can check out 1.2 (still in beta). It has a local mode which helps your setup.

**Reply by BubsonGH** [reply]
Ah nice, Thanks for the heads up, what does this local mode do exactly?

**Reply by apohl79** [reply]
It attaches the plugin window to the AG window and forwards keystrokes into the DAW process. Together with sandboxing enabled, you can also open parallel plugin windows.

**Reply by BubsonGH** [reply]
Interesting, Ill get to downloading. can you elaborate on what sandboxing is? 

Thanks

**Reply by apohl79** [reply]
For each AG plugin the server will spawn a dedicated sandbox process rather than a thread. So a crashing plugin will only kill the sandbox, not the entire server process and thus only affecting a single AG plugin.

**Reply by BubsonGH** [reply]
ahhh I see, yeah I had a couple of drop outs so I can see how that would be very helpful.

Thanks for the help

**Reply by BubsonGH** [reply]
Just tried the beta and its exactly what i needed in regards to opening plugin automatically.

Thanks for letting me know

**Comment by belovedmike** [comment]
I'm curious to know why you got such better performance from the plugins when they were used in this way, as it kind of feels that anyone using AA plugins (or any CPU heavy plugins) should probably be doing the same thing. It sounds like by using AudioGridder, despite being on the same machine, you are able to use more cores of your CPU? I would expect in 2021, especially for something like my M1 Mac Mini, that the CPU is intelligent enough to distribute the load across all cores to maximise resources, is that not true? Or is it down to your DAW? I'm using Studio One 4.6 and although I can load many instances of plugins, responsiveness of the DAW suffers, and I do occasionally get spikes in CPU. Should I be going down the AG route on the same machine (not currently planning to drop another £800 on a dedicated Mac Mini for this)?

Finally, did you need to partition your hard drive to install or is as simple as installing it on the same system?
Thanks,
Mike

**Reply by BubsonGH** [reply]
Hi Mike, 

Yeah it is mostly down to the DAW itself. It does use all of the cores of the CPU but only in the sense of spreading each track across different cores. Any one single track that has a bunch of plugins stacked on it will only be able to use 1 core for all of the plugins on that one track. Which of course can bump up the usage real quick with heavy plugins such as acustica. Using AG for each plugin on a given track sends that one plugin out to the AG server (in my case installed on the same system) and in effect each plugin is going out to the server on its own and being placed on any given core that suits it. in effect allowing any one track to spread its load across all cores.

Hope that makes sense.

**Reply by belovedmike** [reply]
Yep got it, makes much more sense now. Ok thanks very much for taking the time to explain it, much appreciated. I may give it a try, nothing to lose. You didn't say whether you partitioned your drive but I assume there's no need and it's a simple install

**Reply by BubsonGH** [reply]
Ah yeah sorry, in regards to partitions there is no need. its as simple as installing the server software and vst on the same system just like you would anything else. the plugin will see the software running outside of the daw and do what it needs to automatically. all very simple tbh.

I would also recommend installing latest beta and it has some local mode options you may want to try out.

**Reply by belovedmike** [reply]
Ace. Thanks Dan, will do

**Comment by bharathj63** [comment]
In Ableton i still see the performance meter at the bottom of every track spiking. Does Ableton handle it differently?
